src/arithmetic/arithmetic_expression.c

```diff
@@ -145,6 +141,7 @@ static AR_ExpNode *_AR_EXP_CloneOp(AR_ExpNode *exp) {
 	const char *func_name = exp->op.f->name;
 	uint child_count = exp->op.child_count;
 	AR_ExpNode *clone = AR_EXP_NewOpNode(func_name, child_count);
+	if(clone->op.f->callbacks.clone) clone->op.private_data = clone->op.f->callbacks.clone(exp->op.private_data);
```

Comment:
```c
clone = clone->op.f->callbacks.clone;
void *pdata = exp->op.private_data;
if(pdata != NULL && clone != NULL) {
    // clone callback specified, use it to duplicate function's private data
    clone->op.private_data = clone(exp->op.private_data);
}
```

Reply:
in case of distinct we can't check if pdata is not null